addons/mail/models/res_users.py

```diff
@@ -301,11 +312,23 @@ def systray_get_activities(self):
                 "planned_count": 0,
                 "view_type": getattr(Model, '_systray_view', 'list'),
             }
-            for record, activities in activities_by_record.items():
-                if record not in allowed_records:
-                    continue
-                for activity in activities:
-                    user_activities[model_name]["%s_count" % activity.state] += 1
-                    if activity.state in ("today", "overdue"):
-                        user_activities[model_name]["total_count"] += 1
+            if model_name == 'mail.activity':
+                user_activities[model_name]['activity_ids'] = activities.ids
+            for activity in activities:
+                user_activities[model_name]["%s_count" % activity.state] += 1
+                if activity.state in ("today", "overdue"):
+                    user_activities[model_name]["total_count"] += 1
+        if "mail.activity" in user_activities:
+            user_activities["mail.activity"].update({
+                "name": _("Other activities"),
+                "view_id": self.env.ref('mail.mail_activity_view_tree_readonly').id,
+                "view_type": "list",
+            })
         return list(user_activities.values())
+
+    @api.model
+    def _systray_get_allowed_records(self, res_model, res_ids):
+        try:
+            return self.env[res_model].search([("id", "in", res_ids)])
+        except AccessError:
+            return self.env[res_model]
```

Comment:
Here if I can't read a single record, all activities go into "other activities". You should probably keep ids of returned records, and give the remaining ids to "other activities"

Reply:
Maybe use check_access_rules / check_access_rights instead ... it is clearer than using a search.